fdm-app/app/routes/signin.tsx

```diff
@@ -0,0 +1,146 @@
+import type { ActionFunctionArgs, LoaderFunctionArgs } from "react-router";
+import { redirect } from "react-router";
+
+// Components
+import { Button } from "@/components/ui/button"
+import { Check } from "lucide-react"
+
+// Services
+import { cn } from "@/lib/utils";
+import { signIn } from "@/lib/auth-client";
+import { auth } from "@/lib/auth.server";
+
+export async function loader({
+    request,
+}: LoaderFunctionArgs) {
+
+    // Get the session
+    const session = await auth.api.getSession({
+        headers: request.headers
+    })
+
+    // If user has an session redirect to app
+    if (session?.session) {
+        return redirect("/app")
```

Comment:
questionmark is oke here?

Reply:
Yes, it means: to check session only if session exists. 
You could also write it as a nested if like this:
```
if (sesion) {
   if (session.session) {
       return redirect("/app")
   }
} 
```